build/gulp/ts.js

```diff
@@ -71,4 +74,23 @@ gulp.task('ts-check', ['ts-sources'], function() {
         }, ts.reporter.fullReporter()));
 });
 
-gulp.task('ts', [ 'ts-vendor', 'ts-sources', 'ts-check' ]);
+gulp.task('ts-compilation', function() {
+    return gulp.src(TS_PATH)
+        .pipe(ts({
+            noEmitOnError: true
+        }, ts.reporter.fullReporter()));
+});
+
+gulp.task('ts-up-to-date', function() {
+
+    const current = fs.readFileSync(TS_PATH).toString();
+    shell.exec(`npm run update-ts`);
```

Comment:
- I like the `git status` check more. `npm run` call from `gulp` feels unnatural (IMO)
- With `shelljs` check for exit code:
   ```js
   var result = shell.exec("...")
   if(result.code !== 0)
       throw ...;
   ```